pysmt/fnode.py

```diff
@@ -660,3 +760,8 @@ def _env():
 def _mgr():
     """Aux function to obtain the formula manager."""
     return pysmt.environment.get_env().formula_manager
+
+def _is_bv(node):
+    """Aux function to check if a fnode is a BV."""
+    return (node.is_symbol() and node.symbol_type().is_bv_type()) or \
```

Comment:
OK for now, but we shall consider a "shallow" type-checker that only returns the type of the current node without recursion to check type soundness...